Library/Homebrew/install.rb

```diff
@@ -327,7 +327,7 @@ def install_formulae(
         end
 
         if dry_run
-          if (formulae_name_to_install = formula_installers.map(&:name))
+          if (formulae_name_to_install = formula_installers.map(&:formula))
```

Comment:
Maybe this is better, more explicit that we're wanting the name?

```suggestion
          if (formulae_name_to_install = formula_installers.map { |fi| fi.formula.name })
```